pkg/kubelet/kubelet_pods.go

```diff
@@ -1919,12 +1919,13 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 		s.Conditions = append(s.Conditions, status.GeneratePodReadyToStartContainersCondition(pod, podStatus))
 	}
 	allContainerStatuses := append(s.InitContainerStatuses, s.ContainerStatuses...)
-	s.Conditions = append(s.Conditions, status.GeneratePodInitializedCondition(&pod.Spec, allContainerStatuses, s.Phase))
-	s.Conditions = append(s.Conditions, status.GeneratePodReadyCondition(&pod.Spec, s.Conditions, allContainerStatuses, s.Phase))
-	s.Conditions = append(s.Conditions, status.GenerateContainersReadyCondition(&pod.Spec, allContainerStatuses, s.Phase))
+	s.Conditions = append(s.Conditions, status.GeneratePodInitializedCondition(pod, allContainerStatuses, s.Phase))
+	s.Conditions = append(s.Conditions, status.GeneratePodReadyCondition(pod, s.Conditions, allContainerStatuses, s.Phase))
+	s.Conditions = append(s.Conditions, status.GenerateContainersReadyCondition(pod, allContainerStatuses, s.Phase))
 	s.Conditions = append(s.Conditions, v1.PodCondition{
-		Type:   v1.PodScheduled,
-		Status: v1.ConditionTrue,
+		Type:               v1.PodScheduled,
+		ObservedGeneration: podutil.GetPodObservedGenerationIfEnabledOnCondition(pod, v1.PodScheduled),
+		Status:             v1.ConditionTrue,
```

Comment:
I think it's not safe to assume that pod.Status is up-to-date here, since we don't update the pod cache on every status update (I don't think? I need to confirm). Rather, you should use `oldPodStatus`. You could probably create a shallow-copy and overwrite the status with `oldPodStatus`, but I'd prefer to just plumb the status through to `GetPodObservedGenerationIfEnabledOnCondition` directly (maybe pass the status and latest generation instead?)

Reply:
updated!

Reply:
https://github.com/kubernetes/kubernetes/pull/130650 already has approval, so I'm going to let that merge but that means this one will need to be rebased and those changes re-reviewed

Reply:
rebased